wakuv2/waku.go

```diff
@@ -996,158 +985,17 @@ func (w *Waku) SkipPublishToTopic(value bool) {
 	w.cfg.SkipPublishToTopic = value
 }
 
-func (w *Waku) checkIfMessagesStored() {
-	if !w.cfg.EnableStoreConfirmationForMessagesSent {
-		return
-	}
-
-	ticker := time.NewTicker(hashQueryInterval)
-	defer ticker.Stop()
-	for {
-		select {
-		case <-w.ctx.Done():
-			w.logger.Debug("stop the look for message stored check")
-			return
-		case <-ticker.C:
-			w.sendMsgIDsMu.Lock()
-			w.logger.Debug("running loop for messages stored check", zap.Any("messageIds", w.sendMsgIDs))
-			pubsubTopics := make([]string, 0, len(w.sendMsgIDs))
-			pubsubMessageIds := make([][]gethcommon.Hash, 0, len(w.sendMsgIDs))
-			pubsubMessageTime := make([][]uint32, 0, len(w.sendMsgIDs))
-			for pubsubTopic, subMsgs := range w.sendMsgIDs {
-				var queryMsgIds []gethcommon.Hash
-				var queryMsgTime []uint32
-				for msgID, sendTime := range subMsgs {
-					if len(queryMsgIds) >= maxHashQueryLength {
-						break
-					}
-					// message is sent 5 seconds ago, check if it's stored
-					if uint32(w.timesource.Now().Unix()) > sendTime+messageSentPeriod {
-						queryMsgIds = append(queryMsgIds, msgID)
-						queryMsgTime = append(queryMsgTime, sendTime)
-					}
-				}
-				w.logger.Debug("store query for message hashes", zap.Any("queryMsgIds", queryMsgIds), zap.String("pubsubTopic", pubsubTopic))
-				if len(queryMsgIds) > 0 {
-					pubsubTopics = append(pubsubTopics, pubsubTopic)
-					pubsubMessageIds = append(pubsubMessageIds, queryMsgIds)
-					pubsubMessageTime = append(pubsubMessageTime, queryMsgTime)
-				}
-			}
-			w.sendMsgIDsMu.Unlock()
-
-			pubsubProcessedMessages := make([][]gethcommon.Hash, len(pubsubTopics))
-			for i, pubsubTopic := range pubsubTopics {
-				processedMessages := w.messageHashBasedQuery(w.ctx, pubsubMessageIds[i], pubsubMessageTime[i], pubsubTopic)
-				pubsubProcessedMessages[i] = processedMessages
-			}
-
-			w.sendMsgIDsMu.Lock()
-			for i, pubsubTopic := range pubsubTopics {
-				subMsgs, ok := w.sendMsgIDs[pubsubTopic]
-				if !ok {
-					continue
-				}
-				for _, hash := range pubsubProcessedMessages[i] {
-					delete(subMsgs, hash)
-					if len(subMsgs) == 0 {
-						delete(w.sendMsgIDs, pubsubTopic)
-					} else {
-						w.sendMsgIDs[pubsubTopic] = subMsgs
-					}
-				}
-			}
-			w.logger.Debug("messages for next store hash query", zap.Any("messageIds", w.sendMsgIDs))
-			w.sendMsgIDsMu.Unlock()
-
-		}
-	}
-}
-
 func (w *Waku) ConfirmMessageDelivered(hashes []gethcommon.Hash) {
-	if !w.cfg.EnableStoreConfirmationForMessagesSent {
+	if !w.cfg.EnableStoreConfirmationForMessagesSent || w.messageSentCheck == nil {
```

Comment:
Why do we need to check for both? 
If config is set, messageSentChecker should not be nil right.